### DISCUSSION TITLE
Welcome to ME3255 Winter 2022-23

### DISCUSSION THREAD

**Discussion by cooperrc** [cooperrc/computational-mechanics#137] [General]
Welcome to ME3255!

We are starting a fast-paced computational mechanics course. Let's get to know each other today. 
- _Who_ are you? and _Why_ are you here in ME3255?
- What did you do for the holidays (travel? see friends/family? turn on a generator for the power outage? etc.?)
- Show us an example code in Python and tell us what it does. 
    _use the ` ```python - ``` ` markdown format to show code e.g._

````
```python 
x = 1
print(x)
```
````

becomes

```python
x = 1
print(x)
```

**Comment by cooperrc** [comment]
Hi! I'm Prof. Cooper, I decided to offer Computational Mechanics in the Winter because there were so many students with +18-credit semesters this year so I wanted to help alleviate some Spring-semester stress by offering a quick winter session. I love this course! Working with Python + Engineering is so powerful. You can really do great things. 

This holiday season we celebrated all 8 nights of Channukah (almost ran out of candles) & had x2 Christmas mornings. It has been a marathon of presents+food+family. _On top of that_ we lost power Christmas eve due to the storm and had to move the first Christmas morning to my parents' house in Niantic, phew!

```python
for i in range(1, 20):
    if i%3 == 0 and i%5 == 0:
        print('fizzbuzz')
    elif i%3 == 0:
        print('fizz')
    elif i%5 == 0:
        print('buzz')
    else: print(i)
```

This is a [fizz-buzz](https://en.wikipedia.org/wiki/Fizz_buzz) Python solution. It will print out 20 lines of output. If the line number is divisible by 3 or 5 it will replace the number with 'fizz' or 'buzz', respectively. If its divisible by 3&5, it will print out 'fizzbuzz'. Try it out!

**Comment by jpalchisaca** [comment]
Hi I am Jonathan Palchisaca, I decided to take this class because it was a good way to keep me engaged with classes and it would be a smooth transition into spring semester. For the holidays, I had family over and I ate so much despite the cold.

I have not used python in a while, this is a simple addition equation.  It will print out '2023'.
```python -
x = 2022
y= x +1
print(y)  ```

**Reply by cooperrc** [reply]
Welcome!

> I have not used python in a while, this is a simple addition equation. It will print out '2023'.

You're not alone, most people I ask about their Python use say, "I used it 2-3 years ago in CSE 1010 and then never again." Hopefully, you'll find some ways to use programming in your other classes.

**Comment by C-Doll** [comment]
Hello! 

My name is Christian Doll and I am in ME3255 this winter because it is a requirement for ME majors. I have never learned python but I have taken C++ as an equivalent course. For the holidays my wife and I visited our families. Hers on Christmas eve, mine on Christmas day.

Here is some python code.


```python 
length=8
width=10
area=length*width
print('The area is:', area, 'mm squared')
```

This code gives parameters to length, width and area before outputting "The area is 80 mm squared"

**Reply by cooperrc** [reply]
Welcome!
> I have never learned python but I have taken C++ as an equivalent course. 

Python is written in C/C++ so there are tons of more advanced Python use-cases that let you custom-design C-code to speed up Python work. We're going to stick to basics (scripts + functions), but your C++ background can be a real advantage in this case.

**Comment by rneil02** [comment]
Hi! My name is Raye Neil and I decided to take ME3255 this winter to help make my spring semester schedule lighter. For the holidays I  saw some of my extended family, and now my family is visiting Montreal for a few days!  The code below prints the solution to a quadratic equation. 

```
a=1
b=2
c=3
sol1 = ((-b + (b ** 2 - 4 * a * c) ** 0.5) / (2 * a)
sol2 = ((-b - (b ** 2 - 4 * a * c) ** 0.5) / (2 * a)
print('The solutions are' sol1 'and' sol2)

 ```

**Reply by cooperrc** [reply]
> visiting Montreal

Tres Bon!

**Comment by zyhuggan** [comment]
Hello! My name is Zyaja Huggan. I am in ME3255 because it applies a lot to the research I am currently doing and it also helps a lot with my future work load.  For the holidays, I watched many movies with my family and baked desserts. I also spent time with my family from New York. 

The concept of the code below is to recursively determine the amount someone has in their bank account after they have paid a certain balance that they owe. The code keeps track of whether the person has enough to pay the amount, if they successfully paid the amount, and if they overpaid. 
``` python
def payment(paid, current_balance, amount_owed = 100):
    if current_balance <= 0:
        return "Too Much"
    elif paid == amount_owed: 
        return current_balance 
    elif paid > amount_owed:
        return "Overpaid"
    else:
        return payment(paid + 10, current_balance - paid)
```

**Reply by cooperrc** [reply]
> I am in ME3255 because it applies a lot to the research I am currently doing and it also helps a lot with my future work load

Cool! What kind of research do you do?

**Reply by zyhuggan** [reply]
I am doing research on how mutations affect the mechanical properties of different proteins.

**Comment by chrischaplinsky3** [comment]
Hi, my name is Chris Chaplinsky. I am in ME 3225 because it is required for mechanical engineering majors and is a good way to use my time to get ahead for the future. I spent Christmas Eve and Day with my family, eating food and enjoying company. I haven't used python in forever, so this is a simple code that outputs the three-point shooting percentage for any basketball player.

```
a = 82
m = 34
percentage = (m/a)*100

print('Three Point Percentage is ', percentage, '%')'
```

**Reply by cooperrc** [reply]
Welcome!

> I am in ME 3225 because it is required for mechanical engineering majors and is a good way to use my time to get ahead for the future

There are so many different course schedules for people right now. I think we're still feeling effects from a 2-year global quarantine. I hope this course helps alleviate a little Fall/Spring stress.

**Comment by mjaisevbonaye** [comment]
> Hello! My name is Michael Aisevbonaye. I'm taking this course partly because it's an ME grad requirement and it makes my last semester a little easier, but also because I love mechanics and want to take a stab at related problems from a computational frame to help with some projects I'm working on in the spring. 

> Our holiday festivities were a bit smaller due to weather-based travel cancelations and positive covid tests, but unfortunately that didn't stop us from making a truly ridiculous amount of food. I'll be spending the rest of break struggling to make my way through it -_-

> 
> ```python
>def line(x1, y1, x2, y2):
>   m = (y2-y1)/(x2-x1)
>   b = y1 - m*x1
>   if b == 0:
>      print("y = "+str(m)+"x")
>   elif b < 0:
>      print("y = "+str(m)+"x"+" - "+str(-1*b))
>   else:
>      print("y = "+str(m)+"x"+" + "+str(b))
> ```

> The above code returns the line that connects two points in slope-intercept form.

**Reply by cooperrc** [reply]
> Our holiday festivities were a bit smaller due to weather-based travel cancelations and positive covid tests, but unfortunately that didn't stop us from making a truly ridiculous amount of food

Woof, we definitely had a lot more flu/covid/travel changes here too. Glad you still had time to break bread with family!

**Comment by eroot9** [comment]
Hi! My name is Emily Root. I am taking ME 3255 because I really enjoy computer science and I think that it will be very interesting to see more about how I can use it for mechanical engineering purposes. For the holidays this year, I spent Christmas Eve with my dad’s side of the family and Christmas Day with my mom’s side of the family. It was really nice to see a lot of people that I haven’t been able to see in a while!

Below is code for a game where you have to guess the correct number. The game randomly chooses an integer between 1 and 10 and you can guess as many times as it takes to guess the correct number. The program will give you feedback on whether your guess is too high or too low, and it will also tell you when you are correct.

```python 
import random
num = random.randint(1, 10)
guess = int(input("Guess a number between 1 and 10:"))
correct = 0
while correct == 0:
    if guess > num:
        print('Too high.')
        guess = int(input("Guess a different number:"))
    elif guess < num:
        print('Too low.')
        guess = int(input("Guess a different number:"))
        correct == False
    else:
        print('You win!')
        correct += 1
```

**Reply by cooperrc** [reply]
>  The program will give you feedback on whether your guess is too high or too low, and it will also tell you when you are correct.

Awesome! My average score is 4 now. I wonder if there's an optimal starting number to reduce number of guesses?

```python
Guess a number between 1 and 10: 7

Too high.

Guess a different number: 5

Too high.

Guess a different number: 1

You win!
```

**Comment by coryhemsen** [comment]
(repost because my first code didn't work oops!)
Hi! My name is Cory Hemsen. I am taking ME 3255 over the winter to lighten my course load for my last semester at UCONN. I have also not used python in some time, so it will be nice to get a refresher and learn some new material. 

I had a busy holiday season visiting family around the state.

Below is a Python code that calculates GPA based on letter grade according to the UCONN grading scale.

```python
def calculate_gpa(grade1, grade2, grade3, grade4):
 
  grade_values = {
    'A': 4.0,
    'A-': 3.7,
    'B+': 3.3,
    'B': 3.0,
    'B-': 2.7,
    'C+': 2.3,
    'C': 2.0,
    'C-': 1.7,
    'D+': 1.3,
    'D': 1.0,
    'D-': 0.7,
    'F': 0.0
  }

  total_gpa = (grade_values[grade1] + grade_values[grade2] + grade_values[grade3] + grade_values[grade4]) / 4

 
  return total_gpa


print(calculate_gpa('A', 'B+', 'A-', 'B-')) 
```

**Reply by cooperrc** [reply]
> (repost because my first code didn't work oops!)

Nice find! I've posted so many mistakes. 

**PSA:** You can edit your posts with the drop down menu `...` in the top-right of your post, 

**For example  ->**


![image](https://user-images.githubusercontent.com/11371428/209705737-b574c60e-afad-40f4-b67c-650886cbdfc7.png)

**Comment by aliciasle** [comment]
Hello! I am Alicia Sleight, and I am a senior studying BME and ME at UConn.  
I am taking this course so I can lighten my schedule next semester.  The last time I used Python was freshman year, so this course will be a good refresher.
For the holidays, I had lots of family over (and all of their dogs).  The dogs outnumbered the humans! It was wild!

My code outputs a countdown to new years day!

```python - 
import datetime

def new_years_countdown():
    now = datetime.datetime.now()
    new_year = datetime.datetime(now.year + 1, 1, 1)
    days_until_new_year = (new_year - now).days
    hours_until_new_year = days_until_new_year * 24
    minutes_until_new_year = hours_until_new_year * 60
    seconds_until_new_year = minutes_until_new_year * 60
    return f"There are {days_until_new_year} days, {hours_until_new_year} hours, {minutes_until_new_year} minutes, and {seconds_until_new_year} seconds until New Year's Day!"
print(new_years_countdown())

**Reply by cooperrc** [reply]
Nice use of the `datetime` library. Dates+times can be _very_ confusing to nail down specifics. 

One useful feature is that a `datetime` object can be added/subtracted e.g. 

```python
now = datetime.datetime.now()
datetime.datetime(2022, 12, 31, 12, 59, 59) - now
print(str(cntdown), 'hours:min:sec.us')

out: 3 days, 23:34:23.201714 hours:min:sec.us
```

**Comment by mhansen1226** [comment]
Hi, my name is Matthew Hansen. I am a Mechanical Engineering and Engineering & Management gement student at Clarkson University and I am using this class to fulfill my ME requirement for Numerical Methods. If I didn't take it now I would have to take it in overlapping time slots with Heat Transfer and that sounded like no fun, so here I am! I am very glad that Uconn offers this course in the winter. I spent the holidays with my family and ate lots of good food. Here is an example of a simple way to generate the fibonacci sequence with recursion.

```python
def fibonacci(n):
    if n == 0 or n == 1:
        return n
    return fibonacci(n-1) + fibonacci(n-2)

for i in range(10):
    print(fibonacci(i))
```

**Comment by Lordhannnn** [comment]
Hi, my name is Lord Han Godric Magpali. I am a senior studying Mechanical Engineering and am taking this class to reduce the credit I have to take next semester so that I don't have to wake up super early for class :D.  I spent the holidays with my family and friends that came over.
```python
msg = ('Hello World')
print (msg)
```
This is one of the first code that I vividly remember when I took computer science back in high school

**Comment by Nadinemm5** [comment]
Hi I'm Nadine. I'm a third year mechanical engineering major. I decided to take this class now because I knew I would have more time this winter than later on. I planned on taking it for the requisite fulfillment and I also would like to become more proficient in Python for person and professional purposes. I am looking forward to taking another class with Prof. Cooper as well. 

For the holidays I made beef birria tacos for Christmas eve lunch in Thomaston, went to my sisters in New Haven for roast dinner, and made lasagna for Christmas day dinner at home in Bristol. 

My python example I learned in sensors and data when covering confidence intervals and t tests. This example provides an average with a confidence interval for a students grade in a semester.

```
import numpy as np
fall_grades =  [98, 95, 92, 88, 92]
n = 5
v = n-1
t_v95 = 2.776
mean_fall = sum(fall_grades)/n

std_fall = np.std(fall_grades, ddof=1)

ster_fall = std_fall/n**0.5

int_fall = t_v95*ster_fall
print('mu_fall = {} +/- {} (95%)'.format(mean_fall, int_fall))
```

**Reply by cooperrc** [reply]
> For the holidays I made beef birria tacos for Christmas eve lunch in Thomaston, went to my sisters in New Haven for roast dinner, and made lasagna for Christmas day dinner at home in Bristol.

[Beef birria](https://themodernproper.com/beef-birria-recipe) tacos sound amazing :yum: 

> intervals and t tests

FYI: you can import the [t-interval integrals](https://www.statology.org/how-to-find-the-t-critical-value-in-python/) with Python e.g. 

## Find Left-tailed test with 0.05 significance of 0.05

```python
import scipy.stats

#find T critical value
scipy.stats.t.ppf(q=.05,df=22)

-1.7171
```

**Comment by robert-avena** [comment]
Hello everyone, 

My name is Robert Avena, I'm a senior mechanical engineering major. I'm here because I was taking an 18-credit semester in the fall and it was too much! Thanks for offering the course in the winter, Professor Cooper.

For the holidays I headed home to spend some time with my family and caught up with some old friends. I'm going to see the new Avatar movie later today. Has anyone seen it yet? I've heard it's a long one. 

Here's a snippet of a (questionably petty) program I wrote freshman year to see who wasn't following me back on Instagram: (Instagram doesn't let you see that by default and I didn't know how to automate grabbing the usernames, so the program involved copying and pasting a list of names from the website.)

```python 
print("\n\nPeople who don't follow you back:\n\n")

i1 = 0
i2 = 0

for a in _following:
    if a not in _followers:
        i1 += 1 
        print(i1, '@', end='')
        print(a)

print("\n\nPeople who you don't follow back:\n\n")

for b in _followers:
    if b not in _following:
        i2 += 1
        print(i2, '@', end='')
        print(b)

```
This piece of code compares the two lists to see which names are missing from one or the other. If a name is missing, it prints it along with a number (thus the i1/i2 index) so you can count how many there are. Very easy with python.

**Reply by cooperrc** [reply]
Welcome! glad this will lighten the Spring semester load. 

> I'm going to see the new Avatar movie later today. Has anyone seen it yet?

I saw the first one when I was in grad school. It was cool to see it in the theaters. I've heard good things so far. How was it?

**Comment by Sam-Besch** [comment]
Hello, I'm Sam Besch and am a junior studying mechanical engineering. I am taking this winter course in order to lighten my workload later on this spring. For the holidays I've enjoyed spending time with my family watching movies, and seeing friends. I hope to ski and play pond hockey a decent amount this break 

```python 
def add(x, y):
sum = x + y
print(sum)
```

The code above defines an add function which takes in two values and prints the sum of the two values.

**Comment by andrewzakh1** [comment]
Hello, my name is Andrew. I'm currently a senior in mechanical engineering at UCONN. I chose to take this class during the winter period because I would have more time to focus solely on this course. I haven't done much over break, mostly skiing with my family and working on my senior design project. 

```python
import math
a = 3
b = 5
c = math.sqrt((a**2) + (b**2))
print('c =', c)
```
My code performs the Pythagorean Theorem on a triangle with known 'a' and 'b' values, and solves for 'c'.

**Reply by cooperrc** [reply]
FYI- The Python `math` library can be useful for doing simple calculations, but the [NumPy](numpy.org) math functions are _way_ more extensible. 

If you had 50 triangles with sides `a` and `b`, we could find `c` for all 50 with one line of code e.g. 

```python
a = np.linspace(1, 10, 50) # make side a go from 1 to 10 with 50 steps
b = np.linspace(10, 20, 50) # make side b go from 10 to 20 with 50 steps
```

now, calculate all 50 `c` values, 

```python
c = np.sqrt(a**2 + b**2)
```

![image](https://user-images.githubusercontent.com/11371428/210089999-9790bec3-194d-4e34-96de-37b3243a4783.png)

**Comment by DJA17** [comment]
Hi everyone!

My name is Daniel. I am currently a junior studying ME. I decided to take this course as I would like to get ahead on the ME course curriculum. Over these holidays, I celebrated Christmas with just my close family. It was nice to just relax a little before getting back into school work. The following code defines a function that prints the factorial of a number.
```python
def factorial(n):
    if n == 0:
        return 1
    else:
        return n * factorial(n-1)

num = 5
print(factorial(num))
```
The example above prints the factorial of 5 which should net 120

**Comment by johnf329** [comment]
Hello, my name is John. I am a junior Mechanical Engineering major here at UCONN. I am taking this course in the winter because I am slightly behind the expected plan of study and want to catch up without overburdening myself during the normal semester. I am excited to take this course because I like to play around in python sometimes and create mini projects, albeit slightly nervous about the pace of the class. My holiday was nice, I just got together with the local family, we met at my aunts house for Christmas. 

Below is a very simple calculator to estimate the amount of ornaments that can fit on a Christmas tree of any dimensions.

```python 
# This code will estimate the number of ornaments that can fit on a christmas tree.
# Assumptions made are listed as follows:
#     Each ornament occupies 36 square inches of surface area
#     The tree is a perfect cone
#     All ornaments are hung on the outside of the tree, no layering
#     All Measurements are in inches

Import numpy as np

height = int(input('Enter the height of the tree: '))
bottom_diameter = int(input('Enter the base diameter of the tree: '))
r = bottom_diameter/2

area = np.pi*r*(r+(np.sqrt((height**2)+(r**2))))
num_orn = round(area/36)

print('the maximum estimated number of ornaments is', num_orn)
 ```

**Comment by chfdagf** [comment]
Hi, my name is Cheuk Hei, I am taking this course because I am very interested in this course as I am minoring in Computer Science, and also it helps with the ME prerequisite. I slept a lot during the holidays.

Code below shows all the anagrams for each inputted word if it exists among the given listed words

```python
input_list = []
input_list = [item for item in input("Enter the list items : ").split()] #takes input of strings with spaces as seperators
def find_anagrams(input_set):
    def word_compare(word1, word2='steal'):
        try: #exception handler if input is not a string
            if len(word1) == len(word2):
                counter = 0
                for char1 in word1: #for loop that goes through eaach charcter in word
                    for char2 in word2: #for loop that goes through eaach charcter in word
                        if char1 == char2: 
                            counter+=1 #increment counter if both word have same character in some instance of the string
                if counter == len(word2):
                    return "Anagram" #if counter length equal to word length return that it is an anagram
                else: return (word1,word2)
            else:#if the length of x and y are not equal return tuple
                return (word1,word2)
        except TypeError:#exception handler if input is not a string
            return "Those aren't strings!"
    temp_str = ''
    counter = 0

    for item1 in range(len(input_set)): #loop len of set from (0, length if input)
        temp_list = []
        counter += 1
        for item2 in range(len(input_set)): #loop 2 
            if word_compare(input_set[item1],input_set[item2]) == 'Anagram' and input_set[item1] != input_set[item2]:#if statement to check the item 1 to each iterated item 2 
                temp_list.append(input_set[item2])#add to the temp list if not itself and anagram
        if counter < len(input_set):#checks if at the end of list from counter
            temp_str +=  f"{input_set[item1]}: {temp_list}" +'\n'#if not at end concatenate string with newline
        else:
            temp_str +=  f"{input_set[item1]}: {temp_list}"#concatenate string without new line
    return temp_str#return the string
print(find_anagrams(input_list))
```

**Comment by stanliu88** [comment]
Hello, my name is Stanley Liu, I am a junior in mechanical engineering. I enrolled in this winter course because I want to have more time in my future semesters to explore other classes. For the holiday I am traveling to my grandma's house and staying there for a bit of the break. Much like others in this class, I haven't coded anything since CSE 1010. 

The code below is a simple function that introduces yourself by saying Hello. 

```python - 
def introduction(name):
      return 'Hello I am {}'.format(name)
print(introduction('Stanley'))

**Comment by KevinWan123** [comment]
👋 Hello, my name is Kevin Wan, I am Mechanical Engineering major and Math minor. I am taking this class because I want to reduce my spring semester workload and I like how Professor Cooper incorporate python into his teaching.  For the holidays I am working on building a digital light processing printer for my undergraduate research.

Currently I am learning designing responsive website dynamic websites using react.  I am planning on remaking my javascript website using react: https://kevinwan123.github.io/BitMonopoly/.
```python - 
import openai
import random

openai.api_key = "Enter Your api key"

response = openai.Image.create(
  
  prompt="Computational Mechanics",
  n=10,

  size="1024x1024",  
)
image_url = response['data'][0]['url']
```


This python code uses the open ai api DALL E to create an AI image. I have been using this and ChatGPT extensively over the pass month for a project of mine.  I thought this would be a cool tech to share.

**Comment by Maid-Unknown** [comment]
Hello. My name is Yu Xie. I'm one of the junior students in the ME major. I am taking this course because I realized that I must learn more engineering mathematics methods. I hope to learn a lot from this course. This winter break I stay at home and learned something about dynamic control because I think that is more important in recent compared with the structure design. but I don't have much experience with coding.  I will learn that!

```python 
def print_hi(name):
    name=str(name)
    print("Hi.My name is",name,"and nice to meet you today")

if __name__ == '__main__':
    print_hi('YuXie')

 ```

**Comment by lilawhalen** [comment]
I'm Lila Whalen, and I am taking this course because I'm an ME major and wanted to have a lighter schedule during the semester. We lost power right before Christmas Eve and thanks to Christmas magic maybe (?) we got it back in time to have family over on Christmas Eve, and then we visited family for Christmas Day. I haven't used python much but I know that I can covert a weight in kg to lbs by this function:
weight_kg = input()
weight_lbs = 2.2*weight_kg
print('in lbs,' weight_kg 'is,' weight_lbs)

**Comment by dylanbrochard** [comment]
Hi everyone I’m Dylan Brochard, and I am taking this course to sharpen my coding skills and fulfill elective requirements before I graduate in May. 

For the holidays I typically go home to visit my parents, but this year I decided to stay near Storrs to work.

Python has never been my strong suit, but here is a simple code to calculate how many 8 oz glasses of water you should drink per day


```python 
Water glasses needed = 8

**Comment by darienlloyd** [comment]
Hey everyone, my name is Darien Lloyd and I'm a senior mechanical engineering major. This year for the holidays I went home and all of my family came over. I'm not the strongest with Python but I'm excited to use it more as it would be very helpful. I'm taking this class in order to graduate next semester. This python code calculates the average of a list of numbers:

```python 
def calculate_average(numbers):
total = 0
for number in numbers:
total += number
average = total / len(numbers)
return average
numbers = [1, 2, 3, 4, 5]
average = calculate_average(numbers)
print(f"The average of the numbers is {average}")

**Comment by Sam19016** [comment]
Hi, my name is Sam Hollworth and I'm taking ME3255 because it is a mechanical engineering major requirement and I wanted to lighten the course load for my future semesters. For the holidays I visited family and friends within CT to spend time having meals and hanging out. I am not very experienced with python but I'm eager to learn more. This code is for determining if a number is a palindrome.
```python 
InputNumber = input()
if InputNumber == InputNumber [::-1]:
    print("palindrome")
else:
    print("not palindrome")
```

**Comment by Ehuyhua** [comment]
Hello everybody I'm Erick Huyhua, and I am taking CompMec to fulfill my ME elective. This would be my third class with Dr. Copper and I can't wait to learn.
During Christmas, I dressed up as Santa and distributed gifts at my family gathering.
This code determines your letter grade from your numerical score.
```python 
def get_letter_grade(score):
  if score >= 90:
    return "A"
  elif score >= 80:
    return "B"
  elif score >= 70:
    return "C"
  elif score >= 60:
    return "D"
  else:
    return "F"
print get_letter_grade(Erick)
```

**Comment by jmcdonnell876** [comment]
Hello everyone my name is Jaden and I am a senior mechanical engineer finally on track to graduate this coming spring. I am here this winter mainly due to the circumstance mentioned by Professor Cooper where I am taking excess credits and taking this over the winter is a huge savior. This holiday season I was able to celebrate with different parts of my family on the days leading up and following Christmas day.

```python 
bs_calls = 32
team_chrips = 56 + bs_calls

tot_chrips = bs_calls + chirps

if tot_chirps > 100:
print("no re entry")

if else:
print("keep chirping")
```

The code above will calculate how how likely I am to get kicked out of the XL center tomorrow dependent on 2 factors including numbers of bs calls and the number of times I decide to yell at he opposing team

**Comment by mahjabinmaria** [comment]
Hello, My name is Maria Mahjabin. I am a junior majoring in biomedical engineering. I am taking this class, so I can get ahead and not have too many classes during my future semesters. I have not worked with python in a long time, so I am excited to brush up on my memory of it. I made a simple code that just calculates the hours in a week.

days.week= 7
hours.day= 24
hours.week= days.week * hours.day

print("There are", hours.week", hours in a week")

**Comment by youngkeen13** [comment]
Hello , my name is Keenan Watson . I am an junior majoring in mechanical engineering. I am taking this class so my future semesters are not as stressful. I have not worked on python for quite some time , so I am excited to learn new material and also get a refresher on what I forgot. I made a code the prints the name you choose

name = 'Keenan"

print( 'Hello my name is ',name)

**Comment by willgh2** [comment]
Hi, my name is Will Hoffman and I'm a junior in mechanical engineering. I'm taking this class to get ahead for some other semesters coming up so I can lighten my workload. For the Holidays I've spent most of my time with friends and family just relaxing at home. This python code checks if a number is even or odd depending upon the input.

```python 
num = int(input("input number: "))
if (num % 2) == 0:
   print("{0} is even".format(num))
else:
   print("{0} is odd".format(num))

**Comment by avelineim** [comment]
Hello All,

I'm Aveline (like javelin), and I'm a junior ME major from Mystic, CT. I'm excited for the opportunity to take this crash course to fill the gaps in my knowledge when it comes to coding and applying computational analysis to the engineering problems I'll encounter in my courses and research, and happy to have another course checked off as I plan out my future semesters.

So far over the break, I've been paying back a little of my sleep debt from the semester by knocking out 12-16 hours every night, deep cleaning my house, going on long adventure walks with my puppy, and had a small Christmas celebration with my mom and brother this past weekend.

It's been a little while since CSE 1010 and I haven't used much Python since, so I've been a little rusty starting up. Here's an attempt at a simple budgeting tool (for my own good after going a little crazy with retail therapy after receiving some gift cards this holiday season):

```python 
#ask user what their income is and print the number
def __init__(self):
   self.income = float(input("How much is your income this week?\n"))
   print("Your total income is $",self.income)
#tells user how much they can spend, which is thirty percent of total income.
   self.spend = self.income*0.3
   print("\nYour total spending allowance is $",self.spend)
```

This tool asks you for your income that week, then tells you how much you can spend based on a fixed proportion of 30%.

**Comment by Awasthi-Adi** [comment]
Hello! My name is Aditya Awasthi; I am a junior mechanical engineering undergraduate student here at UConn. I am taking ME3255 to get a bit ahead in my ME core coursework so that I can pick up a CSE minor my senior year.

I took 2 finals on that Saturday of finals week and then promptly packed my bags to go to North Carolina on Sunday. I was there until yesterday visiting a newborn relative; he is super cute! A very relaxing trip; I recommend everybody takes a while to decompress before the semester starts again.

```python 
class Entry:
    def __init__ (self, item, priority): #Where priority is a list of priorities
        self.item = item
        self.priority = priority

    def __gt__(self, other):
        if len(self.priority) > len(other.priority):
            return True
        elif len(self.priority) < len(other.priority):
            return False
        else:
            for i in range(len(self.priority)):
                if  self.priority[i] > other.priority[i]:
                    return True
                elif self.priority[i] < other.priority[i]:
                    return False
        return False

    def __eq__ (self, other):
        if self.item != other.item:
            return False
        if len(self.priority) > len(other.priority) or len(self.priority) < len(other.priority):
            return False
        else:
            for i in range(len(self.priority)):
                if  self.priority[i] != other.priority[i]:
                    return False
        return True
```
Above is a code I wrote to define an entry for a type of heap data structure; the entry class also supports functions that check if one entry is greater than another or equal to another entry; not entirely sure if it works. Best of luck to everyone in the course!

**Comment by cgulish** [comment]
Hi! My name is Chris Gulish, and I am a senior Biomedical Engineering major. I am taking this class in order to graduate on time this May. I have barely used python before, and being a transfer student, I never officially took CSE 1010 (Gateway's equivalent taught basic MATLAB). Christmas was very nice, and I was just happy to be home and spend it with family after a very long semester. Below is a simple code that involves addition.

```python 
x = 33
y = 11
z = 6
print(x+y+z)
```

**Comment by seancari1001** [comment]
Hi! My name is Sean and I am a junior here at UConn. I decided to take the class in order to satisfy a requirement, and also I had prof Cooper in the past, and he has been great! I also like the style of the class, so that's one of the reasons why as well.

In terms of the holidays, it was nothing special, just an average Christmas. Went to see some relatives, so that was nice.

If the keys in the dictionary are even… “If the keys are even this will print” will print

```python
cooldictionary = {1:'Chicken',2:'If',3:'Toad',4:'The',5:'Lettuce',6:'Keys',7:'Apple',\
8:'Are',9:'Rain',10:'Even',11:'Smoke',12:'This',13:'Energy',14:'Will',15:'Soup',16:'Print'}

for i in range(1, len(cooldictionary)+1):
    if i%2==0:
        print(cooldictionary[i])
```

**Comment by cgallagher1021** [comment]
Good morning everyone! 
My name is Conor, I'm a senior at UConn .I'm taking this class to fulfill one of my final 2 requirements before graduating this Spring. 
For the holidays I enjoyed a couple different family events and now I am back in the UConn area for work during the week (I'm not sure if anyone else lost power last night, but the UConn area was completely down.)
The code below is a simple function using Python. This is my first time ever using Github so wish me luck. 
```python

> x = 3 

> z = 30

> y = 20

> p = x * y - z

> print(p)
```

**Comment by AmanuelDagnew** [comment]
Good afternoon,

My name is Amanuel, I am senior mechanical engineering student at Indiana Tech. I am taking this class to graduate on time. As an ME student the only coding class I took was math lab. I have used python to do quadratic equations for dynamics class. For the holidays, I travelled to Saint Louis to visit my girlfriend's family.

import cmath  
a = float(input('Enter a: '))  
b = float(input('Enter b: '))  
c = float(input('Enter c: '))  
  
# calculate 
d = (b**2) - (4*a*c)  
  
# find two solutions  
sol1 = (-b-cmath.sqrt(d))/(2*a)  
sol2 = (-b+cmath.sqrt(d))/(2*a)  
print('The solution are {0} and {1}'.format(sol1,sol2)) 

I used this to save time while doing projectile motion problems. A, B and C are given.

**Comment by ElisDiaz46** [comment]
My name is Elis Diaz. I am taking ME 3255 because I am interested in applying math into Computer Science. What I did in the Holidays. I went to church and had a good time and later went home and relax. 

X = 'Elis'
Y = 'Diaz'
Name = X " " Y
Print(Name)

**Reply by cooperrc** [reply]
There's a couple little corrections here, 

```python
X = 'Elis'
Y = 'Diaz'
Name = X + " " +  Y # strings can be added to each other, but 
                                  # if you separate by spaces, you get a Syntax error
print(Name) # Python is case sensitive, make sure `print` is lowercase
```

**Comment by pj931** [comment]
Hi,
My name is Peter Leppones. I am a senior in electrical engineering.

I am in this course to learn more about using python to solve engineering problems and visually represent data.

Here is some code to roll a die:

```python
import random

print(randint(1,6))

```

**Comment by JonCarlosRestrepo** [comment]
Hello, my name is Jon Carlos Restrepo Jr. I am currently attending this summer course to overall make my last year at UConn as a future senior smooth and lighter. 

During the summer, I had taken the opportunity to get into study abroad at Prague for three weeks and after I'm going to get into my internship while taking my summer courses.

The code below is to show that I like doughnuts

```python 
a = 'I'
b = 'like'
c = 'Doughnuts'
Name = a + " " + b + " " + c
print(Name)
 ```
